### ISSUE TITLE
Game "Freezes" When Walking Around in Town and Cities

### ISSUE THREAD

**Issue by Cpl-Jordan** [CleverRaven/Cataclysm-DDA#46653]
# Describe the bug
What this bug is when I move my character in a town or city the game freezes for 3-5 seconds when I move my character 14-15 tiles. (It seems to freeze when my character walks between the street lamps on a roadway any direction)

# Steps To Reproduce
1: Head into any dense (Dense as in buildings) town/city in the game
2: Move your character in one direction and pass a street lamp and the game will then freeze for 3-5 seconds.

# Expected behavior
I expect being able to walk though the town/cities roadways without any freezing in the game. Lets say I clear out a long roadway from all monsters so I can walk down the road. When I do walk down the road the game freezes every 14-15 tiles. (I apologize if this sounds rude I don't mean to be at all I am just following the template) What I expect the cause of this to be is the game loading the new tiles of the map when my character moves into the new area. What has drawn me to this conclusion is I asked the question on the discourse for CDDA and one person on there said they expected it to be the game loading that area of the map. As when I walk back on the same roadway I walked to the game doesn't freeze unless the game saves. If the game saves it freezes on the way back as well. Here is the link to that topic as well if that helps: https://discourse.cataclysmdda.org/t/game-freezes-when-walking-around-in-towns-and-citys/25666

# Screenshots
Here is a screenshot and some things I added onto the screenshot to best describe this bug.
![Bugscreenshot](https://user-images.githubusercontent.com/77216510/104112365-f7854380-52bb-11eb-8ffa-91a036dea99f.png)

# Versions and configuration
- OS: Windows
    - OS Version: 10.0 2004
- Game Version: 0.E-8807-gffd58c6 [64-bit]
- Graphics Version: Tiles
- Game Language: English
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Bionic Professions [package_bionic_professions]

# Additional context
I will provide my save file below. I set my character up on the roadway facing the East. Just walk down that street and the game should freeze every time the character passes between the street lamps as shown in the screen shot.
[CDDA_Save.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/5792091/CDDA_Save.zip)

**Comment by kevingranade** [comment]
This issue has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/game-freezes-when-walking-around-in-towns-and-citys/25666/11

**Comment by Cpl-Jordan** [comment]
> This issue has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:
> 
> https://discourse.cataclysmdda.org/t/game-freezes-when-walking-around-in-towns-and-citys/25666/11

Oh, I was the poster of that discussion. I was asked to file out a bug report and upload my save on here. I don't want to come off like I am spamming you guys with this issue.

**Comment by BrettDong** [comment]
It's an automatic comment whenever a link to GitHub issue is posted on discourse forum.

**Comment by Cpl-Jordan** [comment]
> It's an automatic comment whenever a link to GitHub issue is posted on discourse forum.

Hahaha thank you for telling me that. Wow do I look like a dumb dumb now.

**Comment by BrettDong** [comment]
Confirm encountering periodic freeze moving every 12 tiles. Profiling result:
<img width="1552" alt="螢幕截圖 2021-01-09 下午9 13 26" src="https://user-images.githubusercontent.com/21075502/104112655-88115300-52bf-11eb-982a-d3933060d6d4.png">

**Comment by Qrox** [comment]
> every 12 tiles

Sounds like something went wrong with submap loading.

**Comment by BrettDong** [comment]
I feel it's like submap being loaded repeatedly when it should not.

**Comment by BrettDong** [comment]
Another profiling on the 200ms lag, however, looks different:
<img width="1552" alt="螢幕截圖 2021-01-09 下午11 19 07" src="https://user-images.githubusercontent.com/21075502/104114314-1510d800-52d1-11eb-86f0-2e6711549e99.png">

**Comment by Aivean** [comment]
The problem is item pocket deserialization on submap loading, same as in #45822:

![image](https://user-images.githubusercontent.com/2865203/104392476-f68c2600-54f6-11eb-9c10-b27ed802bc81.png)
(with rendering pruned)

**Comment by Aivean** [comment]
Another problem is that `glass_shard` is not stackable and in the city scenario the portion of glass shards among all items could be as high as 60%!

```sh
$ grep -o '"typeid"' 154.16.0.map | wc -l
     816
$ grep -o '"glass_shard"' 154.16.0.map | wc -l
     513
```

**Comment by int-ua** [comment]
Should be fixed by #46763

**Comment by Aivean** [comment]
@int-ua, please don't close this yet. As I explained in https://github.com/CleverRaven/Cataclysm-DDA/pull/46763#issuecomment-760702157, the issue of non-stackable glass shards still remains. I hope to fix it with RLE compression for items stacks in the future PR.
